client/my-sites/site-settings/form-general.jsx

```diff
@@ -698,6 +707,35 @@ export class SiteSettingsFormGeneral extends Component {
 			</div>
 		);
 	}
+
+	advancedCustomizationNotice() {
+		const { translate, siteSlug } = this.props;
+		const upgradeUrl = '/plans/' + siteSlug;
+		const siteUrl = `https://${ siteSlug }`;
+
+		return (
+			<>
+				<div className="site-settings__advanced-customization-notice">
+					<div className="site-settings__advanced-customization-notice-cta">
+						<Gridicon icon="info-outline" />
+						<span>
+							{ translate( 'Your style changes wont be public until you upgrade your plan.' ) }
+						</span>
+					</div>
+					<div className="site-settings__advanced-customization-notice-buttons">
+						<Button href={ siteUrl }>{ translate( 'View site' ) }</Button>
```

Comment:
I think this should open a new tab since the user is leaving Calypso.
```suggestion
						<Button href={ siteUrl } target="_blank">{ translate( 'View site' ) }</Button>
```